File: system-test/sec-tests/java/Sec12HeaderInjection.java
// SEC-12: Header Injection / CRLF
// Test: TC-CR-018
package com.example;

import javax.servlet.http.HttpServletResponse;

public class RedirectService {
    public void redirect(HttpServletResponse response, String url) {
        response.setHeader("Location", url);
    }
}

File: system-test/sec-tests/javascript/sec_12_header_injection.js
// SEC-12: Header Injection / CRLF
// Test: TC-CR-018
const http = require('http');

function handleRedirect(res, url) {
  res.writeHead(302, { 'Location': url });  // Potential CRLF injection if url contains \r\n
  res.end();
}
